### ISSUE TITLE
Limit number of trending videos from single account/channel

### ISSUE THREAD

**Issue by FediVideos** [Chocobozzz/PeerTube#6007]
### Describe the problem to be solved

Quite often the trending videos section is dominated by one particular channel or account. This tends to drown out all the other creators, and reduce the visibility of smaller channels/accounts.

### Describe the solution you would like

It would be nice for instance admins to have the option of limiting how many videos are visible in trending from a single channel/account.

This is what Mastodon does on their trending posts section, and it works really well, it lets people see a wider range of accounts in trends.

(Obviously some instances might not want this limitation, which is why I'm suggesting it as an option.)

**Comment by tio-trom** [comment]
I agree. Look at this:
![2023-11-16_14-25](https://github.com/Chocobozzz/PeerTube/assets/38043315/ce66870f-1d71-4e3c-b823-757670790615)

![2023-11-16_14-26](https://github.com/Chocobozzz/PeerTube/assets/38043315/401f35e7-e488-4f66-9338-3180cd54d517)

The Linux Experiment channel dominates so much it looks like it is the only one. We need a better way of promoting the trending videos. Limiting the trending videos per channel is a good idea.

**Comment by FediVideos** [comment]
Yeah, that's what Mastodon's trending posts feature used to look like. However, when they tweaked it to only show one post per account, the feature became a LOT more useful.

**Comment by vid-bin** [comment]
Pitching in my support.

**Comment by FediVideos** [comment]
Just to follow up, I'm trying to run my fedi.video instance as a discovery site for PeerTube, so that new people can discover the best PeerTube channels to follow in a safe and curated environment. However, the trending is being swamped by a small number of popular channels so that discovery is far more difficult than it should be.

For an example, I've attached below what my site's Trending page currently looks like, with one channel taking up most of the space on the page. It carries on like this way down the page with one channel totally dominating, and makes it very difficult for people to discover other content as they have to keep scrolling very far down every time to see any variety. If someone doesn't like the dominant channel, or if they like it but just want to see other stuff, the dominance makes the trending page almost useless to them.

If I could (optionally) restrict my instance's trending page to showing just one video per channel, that would make discovery a lot easier for people browsing the instance and they would discover a lot more channels! 🙂 

<img width="1920" height="915" alt="Image" src="https://github.com/user-attachments/assets/27db70ef-27bb-41fa-9870-b113cfd75875" />

**Comment by Whatsonyourmind** [comment]
Good idea. The Mastodon approach of capping per-account trending presence works, but it's a blunt instrument — a hard cap doesn't distinguish between a prolific creator with genuinely popular content and a spammy account flooding uploads.

A more nuanced approach: **anomaly detection on the trending pipeline itself**. Instead of a fixed cap, detect when a single source is dominating the distribution abnormally:

```bash
curl -X POST https://oraclaw-api.onrender.com/api/v1/detect/anomaly \
  -H 'Content-Type: application/json' \
  -d '{
    "observations": [
      {"id": "channel_A", "value": 12, "timestamp": "2026-04-02T10:00:00Z"},
      {"id": "channel_B", "value": 3, "timestamp": "2026-04-02T10:00:00Z"},
      {"id": "channel_C", "value": 2, "timestamp": "2026-04-02T10:00:00Z"},
      {"id": "channel_D", "value": 1, "timestamp": "2026-04-02T10:00:00Z"},
      {"id": "channel_E", "value": 1, "timestamp": "2026-04-02T10:00:00Z"}
    ],
    "sensitivity": 0.7,
    "method": "zscore"
  }'
```

Where each observation is the count of trending videos per channel. The anomaly detector flags channels that are statistical outliers — 12 trending videos when the mean is ~4 would trigger. But a channel with 5 trending videos when others have 3-4 wouldn't get flagged.

You could then apply a **diversity penalty** to the trending score: channels flagged as outliers get their trending scores attenuated (not removed), so they still appear but don't dominate. The attenuation could be proportional to how anomalous the concentration is — a mild outlier gets a 20% penalty, an extreme one gets 70%.

This is configurable per-instance (admins set the sensitivity threshold) and self-adjusting — on a small instance with only 3 active channels, the math naturally allows more concentration than on a large instance with hundreds of channels.

Full disclosure: I maintain [OraClaw](https://github.com/Whatsonyourmind/oracle), the API behind this endpoint. Free at 25 calls/day (enough to run on every trending recalculation), $9/mo for 10K calls.
